fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -131,6 +143,70 @@ fun OnboardingPage(
     }
 }
 
+@Composable
+private fun DescriptionText(
+    description: String,
+    linkifiedText: LinkifiedText?,
+) {
+    if (linkifiedText != null) {
+        LinkifiedText(
+            text = description,
+            linkifiedText = linkifiedText,
+        )
+    } else {
+        Text(
+            text = description,
+            color = FirefoxTheme.colors.textSecondary,
+            textAlign = TextAlign.Center,
+            style = FirefoxTheme.typography.body2,
+        )
+    }
+}
+
+/**
+ * A composable for displaying text that has a clickable linkified part.
+ * @param text The complete text.
+ * @param linkifiedText The linkified part of the text.
+ */
+@Composable
+private fun LinkifiedText(
+    text: String,
+    linkifiedText: LinkifiedText,
+) {
+    val annotatedString = buildAnnotatedString {
+        val startIndex = text.indexOf(linkifiedText.text)
+        val endIndex = startIndex + linkifiedText.text.length
```

Comment:
Optional:
```suggestion
        val endIndex = text.lastIndexOf(linkifiedText.text)
```

Reply:
Hmm the edge case with this change would be that `endIndex` could be the `lastIndex` of the subsequent or last occurrence of the string. e.g in "A firefox logo is not really a fox", the part "fox logo is not really a fox" would then be covered by start and end indices, instead of just the first occurrence of "fox". The component doesn't support multiple links but won't error and default to the first occurrence with the existing implementation.